contracts/access/manager/AccessManager.sol

```diff
@@ -439,10 +450,6 @@ contract AccessManager is Context, Multicall, IAccessManager {
      * Emits a {RoleGrantDelayChanged} event
      */
     function _setGrantDelay(uint64 roleId, uint32 newDelay) internal virtual {
-        if (roleId == PUBLIC_ROLE) {
-            revert AccessManagerLockedRole(roleId);
-        }
```

Comment:
Note to self and others: this is removed because the public role is not grantable according to the logic of `_grantRole`.

Reply:
Given that the public role is indeed not grantable, does it make sens to set the grant delay ? IMO it send the wrong signal that you can set a delay, and emit the corresponding event, when you know the delay will never be enforced.

Reply:
Perhaps documentation is fine for the case of granting a delay to the PUBLIC_ROLE, I wouldn't add an extra check for a case that is low likely to happen

Reply:
>IMO it send the wrong signal that you can set a delay, and emit the corresponding event, when you know the delay will never be enforced

This is an interesting point, but this is the grant delay for a role that every account automatically has. Observing `RoleGrantDelayChanged` for such a role should provide no information.

Reply:
> I can't think of any incorrect conclusion someone could draw from this event.

I can imagine someone thinking that new user (if that means anything) are in the grant role with a delay. For example that when a create a new seedphrase/private key (or when I create a new account smart contract) I'll get in the public group after a delay.

Sure that makes no sens, but I'm pretty sure some people out these will believe that.

Reply:
Note that in `_setRoleAdmin` and `_setRoleGuardian` we don't allow setting the admin/guardian for the `PUBLIC_ROLE`. I'm not sure why that would be forbiden, but setting the grantDelay would be authorized.